In [203]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression

In [204]:
# Import the data
train_df = pd.read_csv('/Users/irawobebeyi/Desktop/lending.csv')
# Display the dataframe
train_df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

PREDICTIONS

I strongly believe that the Random Forest Classifier would give a better result than the Logistic Regression Model.



In [205]:
# Split the data into X_train, X_test, y_train,
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(x, y, random_state=42)





In [206]:
#Scaling data based on the values provided
from sklearn.preprocessing import StandardScaler
# Train data
scaler_train = StandardScaler().fit(X_train)
X_train_scaled = scaler_train.transform(X_train)
# Test data
scaler_test = StandardScaler().fit(X_test)
X_test_scaled = scaler_test.transform(X_test)

In [207]:
# Train a Logistic Regression model print the model score
reg = LogisticRegression().fit(X_train, y_train)
reg.score(X_test, y_test)

0.9924680148576145

In [208]:
# Train a Random Forest Classifier model and print the model score

randomForestClass = RandomForestClassifier(random_state=0)
randomForestClass.fit(X_train, y_train)
randomForestClass.score(X_test,y_test)

0.9922100701609575

CONCLUSION

The machine Language algorithims Ramdom Forest Classifier (RFC) and Logistics Regression Model(LFC) are used for different disciplines for classification and regression purposes.
The RFC model score is 0.9922100701609575 and LFC model score is 0.9924680148576145.

This doesnt do justice to my prediction. I predicted that RFC would do better but LFC did better.




